furusiya/model/entities/fire.py

```diff
@@ -0,0 +1,34 @@
+import colors
+from constants import DELTA_UP, DELTA_DOWN, DELTA_LEFT, DELTA_RIGHT
+from game import Game
+from model.entities.game_object import GameObject
+from model.config import config
+
+
+class Fire(GameObject):
+    def __init__(self, x, y):
+        super().__init__(x, y, '*', 'Fire', colors.red, blocks=False)
+
+        Game.events.bind('on_entity_move', self.on_entity_move)
+        Game.events.bind('on_turn_pass', self.on_turn_passed)
+
+        self.turns_passed_alight = 0
+
+    def on_entity_move(self, entity):
+        if (entity.x, entity.y) == (self.x, self.y):
+            Game.fighter_system.get(entity).take_damage(config.data.enemies.fire.damage)
+            self.die()
+
+    def on_turn_passed(self):
+        self.turns_passed_alight += 1
+        if self.turns_passed_alight >= config.data.enemies.fire.selfExtinguishTurns:
+            self.die()
+        if config.data.enemies.fire.spreadChance >= Game.random.randint(1, 100):
+            dx, dy = Game.random.choice([DELTA_UP, DELTA_DOWN, DELTA_LEFT, DELTA_RIGHT])
+            created_fire = Fire(self.x + dx, self.y + dy)
+            Game.area_map.entities.append(created_fire)
```

Comment:
You forgot to check if that tile is a wall (fail) or an entity (which should probably catch fire and instantly extinguish). We need a test for this.

We also need a test for a case where we pick an empty ground tile (nothing should happen).

I would like a test for items (they should burn to ash), even if it fails now.